pkg/lib/cockpit-components-dialog.jsx

```diff
@@ -271,12 +271,14 @@ export function show_modal_dialog(props, footerProps) {
     const dialogName = 'cockpit_modal_dialog';
     // don't allow nested dialogs, just close whatever is open
     const curElement = document.getElementById(dialogName);
+    const current = createRoot(curElement);
     if (curElement) {
-        ReactDOM.unmountComponentAtNode(curElement);
+        current.unmount();
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17939-20221121-134727-dbe868f6-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)